lib/src/model/tournament/tournament.dart

```diff
@@ -0,0 +1,89 @@
+import 'package:deep_pick/deep_pick.dart';
+import 'package:fast_immutable_collections/fast_immutable_collections.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:lichess_mobile/src/model/common/chess.dart';
+import 'package:lichess_mobile/src/model/common/id.dart';
+import 'package:lichess_mobile/src/model/common/perf.dart';
+import 'package:lichess_mobile/src/model/common/time_increment.dart';
+import 'package:lichess_mobile/src/model/user/user.dart';
+import 'package:lichess_mobile/src/utils/json.dart';
+
+part 'tournament.freezed.dart';
+
+typedef TournamentSchedule = ({String freq, String speed});
+
+typedef TournamentLists =
+    ({
+      IList<TournamentListItem> started,
+      IList<TournamentListItem> created,
+      IList<TournamentListItem> finished,
+    });
+
+extension TournamentExtension on Pick {
+  TimeIncrement asTimeIncrementOrThrow() {
+    final requiredPick = this.required();
+    return TimeIncrement(
+      requiredPick('limit').asIntOrThrow(),
+      requiredPick('increment').asIntOrThrow(),
+    );
+  }
+
+  TournamentSchedule asTournamentScheduleOrThrow() {
+    final requiredPick = this.required();
+    return (
+      freq: requiredPick('freq').asStringOrThrow(),
+      speed: requiredPick('speed').asStringOrThrow(),
+    );
+  }
+
+  IList<TournamentListItem> asTournamentListOrThrow() =>
+      asListOrThrow((pick) => TournamentListItem.fromServerJson(pick.asMapOrThrow())).toIList();
+}
+
+@freezed
+class TournamentListItem with _$TournamentListItem {
+  const TournamentListItem._();
+
+  const factory TournamentListItem({
+    required TournamentId id,
+    required String createdBy,
```

Comment:
I'd expect a `UserId` here.

Reply:
It actually contains the username, not the ID. Most notably, for all tournaments of `lichess.org/tournament` it contains `lichess`